src/Symfony/EventListener/ExceptionListener.php

```diff
@@ -30,11 +28,11 @@ final class ExceptionListener
     /**
      * @var ErrorListener
      */
-    private $exceptionListener;
+    private $errorListener;
 
-    public function __construct($controller, LoggerInterface $logger = null, $debug = false, ErrorListener $errorListener = null)
+    public function __construct($controller, LoggerInterface $logger = null, $debug = false)
     {
-        $this->exceptionListener = $errorListener ? new ErrorListener($controller, $logger, $debug) : new LegacyExceptionListener($controller, $logger, $debug); // @phpstan-ignore-line
+        $this->errorListener = new ErrorListener($controller, $logger, $debug);
```

Comment:
Is there anything that prevents injecting the error listener rather than instantiating here? 
```suggestion
        $this->errorListener = $errorListener ?? new LegacyExceptionListener($controller, $logger, $debug); // @phpstan-ignore-line;
```

(+ update of the `api_platform.listener.exception` service definition to give it the `api_platform.listener.error` service as argument)